src/Raven.Client/Server/DatabaseTopology.cs

```diff
@@ -93,55 +94,125 @@ public interface IDatabaseTask
+            list.AddRange(Members.Where(m => m.NodeTag != nodeTag));
+            list.Sort();
+            return list;
+        }
 
-            return AllNodes.Where(n => n != nodeTag).Select(n => new ReplicationNode{
-                NodeTag = n,
-                Url = NameToUrlMap[n],
+        public void AddMember(string nodeTag, string databaseName)
+        {
+            Members.Add(new DatabaseMember
+            {
+                NodeTag = nodeTag,
+                Url = NameToUrlMap[nodeTag],
                 Database = databaseName
-            }).Concat(watchers);
+            });
         }
 
-        public bool MyConnectionChanged(DatabaseTopology other, string nodeTag, string databaseName)
+        public (List<ReplicationNode> nodesToAdd, List<ReplicationNode> nodesToRemove) FindConnectionChanges(DatabaseTopology other, string nodeTag, string databaseName)
         {
-            return GetDestinations(nodeTag,databaseName).SequenceEqual(other.GetDestinations(nodeTag, databaseName)) == false;
-        }
+            var oldDestinations = GetDestinations(nodeTag, databaseName);
+            var newDestinations = other.GetDestinations(nodeTag, databaseName);
+
+            var addDestinations = new List<ReplicationNode>();
+            var removeDestinations = new List<ReplicationNode>();
+            
+            // this will work because the destinations are sorted. 
+            using (var oldEnum = oldDestinations.GetEnumerator())
+            using (var newEnum = newDestinations.GetEnumerator())
+            {
+                newEnum.MoveNext();
+                oldEnum.MoveNext();
+                while (oldEnum.Current != null && newEnum.Current != null)
+                {
+                    if (oldEnum.Current.CompareTo(newEnum.Current) > 0)
+                    {
+                        // add new
+                        addDestinations.Add(newEnum.Current);
+                        newEnum.MoveNext();
+                        continue;
+                    }
+                    if (oldEnum.Current.CompareTo(newEnum.Current) < 0)
+                    {
+                        // remove old
+                        removeDestinations.Add(oldEnum.Current);
+                        oldEnum.MoveNext();
+                        continue;
+                    }
+                    newEnum.MoveNext();
+                    oldEnum.MoveNext();
+                }
+
+                // the remaining nodes of the old destinations should be removed
+                while (oldEnum.Current != null)
+                {
+                    removeDestinations.Add(oldEnum.Current);
+                    oldEnum.MoveNext();
+                }
 
+                // the remaining nodes of the new destinations should be added
+                while (newEnum.Current != null)
+                {
+                    addDestinations.Add(newEnum.Current);
+                    newEnum.MoveNext();
```

Comment:
Same here, should use the result of `MoveNext`, not assume that `Current` is null